### DISCUSSION TITLE
Datepicker doesn't open, class=hidden middlewareData.hide undefined

### DISCUSSION THREAD

**Discussion by markkimsal** [filamentphp/filament#7469] [Help]
I can't quite figure out why my datepicker doesn't open.  If I manually remove the class "hidden" or change the definition of .hidden .d-hide with my browser inspector it shows, but the JS in webpack://filament/packages/forms/dist/module.esm.js doesn't want to change the visibility, the middlewareData.hide is undefined.

This is filament 2, non-admin hand built form.  I just created a DatePicker reference, every thing else about it seems to function.

anybody know why `$x7.$refs.panel.open($x7.$refs.button)` would not remove the class "hidden" from the panel element?  

P.S. I am mixing bootstrap and tailwind :/

**Comment by markkimsal** [comment]
ok, it looks like the date-time-picker blade template sets a class of "hidden", but never removes it.  it only sets the elements local style to "visibility: show".  Well, bootstrap comes with .hidden { visibility: none !important}

so, I either need to add the "x-important" alpine modifier to the date-time-picker.js or just remove the hidden css class.

removing the hidden class works for me, not sure why it's there in the first place. :/

**Comment by danharrin** [comment]
Honestly it sounds like a setup issue, if the class shouldnt be there when we would have had more reports

**Comment by markkimsal** [comment]
Bootstrap 4 uses `display: none !important` for it's definition of .hidden.  tailwind does not use the !important.

I'm probably the only person using bootstrap 4 plus customized tailwind with filament 2 :/

Either adding `.important` to the x-float chain of the date panel _or_ removing the `hidden` class makes it work for me, no flicker or anything.  the `.important` flag makes the local style definition also use an !important, thus overriding the bootstrap !important modifier.

Also, adding this css to my custom css definition "fixes" the "problem" as the panel is a sibling of the button which gets its own aria-expanded property manipulated by the JS

```css
button[aria-expanded=true] ~ div.hidden {
    display: block !important ;
}
```

I think the question remains, why doesn't the panel just toggle the hidden CSS class instead of manipulating element level style declarations?

**Reply by danharrin** [reply]
Because we use the Floating UI package and thats not how it works
